### ISSUE TITLE
Volumes Not Mounting in Dockerized K8S

### ISSUE THREAD

**Issue by nVitius** [kubernetes/kubernetes#16055]
I'm setting up a [docker-multinode](http://kubernetes.io/v1.0/docs/getting-started-guides/docker-multinode.html) k8s. I'm trying to mount a volume (in this case, a secret). Originally, I was getting this issue:
`Unable to mount volumes for pod "service-registry-eccye_default": exit status 32`
I was on Slack yesterday, and @cjcullen pointed out that I should be running the kubelet with the `--containerized` flag. I have added it and restarted the server and now I'm having this error:
`Unable to mount volumes for pod "service-registry-4b682_default": fork/exec /nsenter: no such file or directory`  

It looks like this issue was brought up in the past (#8734), but never resolved.

**Comment by mikedanese** [comment]
Known issue. I think we are waiting for https://github.com/docker/docker/pull/16773 and the issue https://github.com/docker/docker/issues/10088 to converge before we can support this. I think we are tracking this yet though...

@fgrzadkowski

**Comment by nVitius** [comment]
All right, thanks for linking to those. That helps.

**Comment by 2opremio** [comment]
Is this a duplicate of #16586 ? I get a different error.

@mikedanese @fgrzadkowski If it is, can you point me to the changeset which circumvents the problem?

**Comment by 2opremio** [comment]
> Is this a duplicate of #16586 ? I get a different error.

OK, it seems to be a duplicate (I get almost the same error when running `hyperkube:v1.0.7`)

I'm still interested in the specific changeset which works it around in HEAD though.

**Comment by dalanlan** [comment]
secrets thing ref #13791

**Comment by dalanlan** [comment]
and nsenter - - my guess is the following line would do. https://github.com/kubernetes/kubernetes/blob/master/cluster/images/hyperkube/Dockerfile#L13

**Comment by 2opremio** [comment]
@dalanlan Thanks for the pointers! It seems to be fixed in v1.1.0. I will build an image from master and give it a try.

BTW, is there somebody building hyperkube Docker images from master? (gcr.io/google_containers/hyperkube:v1.1.0 nor gcr.io/google_containers/hyperkube:latest exist)

**Comment by dalanlan** [comment]
hyperkube does not release quite often. That said, no `latest` tag to my best knowledge. You could always follow https://github.com/kubernetes/kubernetes/releases anyway. 
Moreover, #13791 has not been cherry-picked into a formal release yet.

**Comment by 2opremio** [comment]
OK, I confirm it works on master.

In case it's useful for someone else, I built my own hyperkube image from master 2opremio/hyperkube                    :664d2ef (see https://github.com/2opremio/kubernetes/tree/kubelet-HEAD-img/cluster/images/hyperkube)

**Comment by brendandburns** [comment]
I will propose a cherry pick of this into 1.1.2 so that we can get a working version from the 1.1.x series.

**Comment by fgrzadkowski** [comment]
This is already fixed for image `v1.2.0-alpha7` and obviously will be working in `v1.2.0`.
